### PR TITLE
[Tensorflow]|[build] Updating TF serving binaries for TF 2.5

### PR DESCRIPTION
**Pull request by hsuw333**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by tejaschumbalkar** [comment]
Modify the `tensorflow/buildspec.yml` file to use `2.5.0` TF version. Remove the entries in `image` related to training builds as well.

File: tensorflow/buildspec.yml [hunk 1]
```diff
@@ -1,16 +1,10 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK tensorflow
-version: &VERSION 2.4.1
-short_version: &SHORT_VERSION 2.4
+version: &VERSION 2.5.1
+short_version: &SHORT_VERSION 2.5
 
 repository_info:
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:7]: `training_repository` part can stay

File: tensorflow/buildspec.yml [hunk 2]
```diff
@@ -19,13 +13,6 @@ repository_info:
       *REPOSITORY_NAME ]
 
 context:
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:15]: `training_context` part can stay as well

File: tensorflow/inference/docker/2.5/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.5/license.txt -o /license.txt
```
Comment (tejaschumbalkar) [inline tensorflow/inference/docker/2.5/py3/Dockerfile.cpu:128]: Does the file exist?

File: tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.5/license.txt -o /license.txt
```
Comment (tejaschumbalkar) [inline tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu:167]: Does the file exist?

**Review by tejaschumbalkar — COMMENTED** [review]
Please confirm if there is any additional test to be added to support TF2.5 with CU112 function ality

File: tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu (hunk 4 above)
Comment (hsuw333) [reply tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu:167]: added the file to the bucket

File: tensorflow/inference/docker/2.5/py3/Dockerfile.cpu (hunk 3 above)
Comment (hsuw333) [reply tensorflow/inference/docker/2.5/py3/Dockerfile.cpu:128]: added file to bucket

File: tensorflow/buildspec.yml (hunk 1 above)
Comment (hsuw333) [reply tensorflow/buildspec.yml:7]: added back

File: tensorflow/buildspec.yml (hunk 2 above)
Comment (hsuw333) [reply tensorflow/buildspec.yml]: added back

**Comment by junpuf** [comment]
@tejaschumbalkar let's use `ami-0542a4236ca7d0a87` for this build, this ami has r460 driver.

**Comment by junpuf** [comment]
us-east-1: `ami-056569e3027c62589`
us-west-2: `ami-0542a4236ca7d0a87`

File: tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+# adds a new list which contains libnvinfer library, so it needs another
+# 'apt-get update' to retrieve that list before it can actually install the
+# library.
+# We don't install libnvinfer-dev since we don't need to build against TensorRT,
+# and libnvinfer4 doesn't contain libnvinfer.a static library.
+RUN apt-get update \
+# nvinfer-runtime-trt-repo doesn't have a 1804-cuda10.1 version yet. see:
+# https://developer.download.nvidia.cn/compute/machine-learning/repos/ubuntu1804/x86_64/
+ && apt-get install -y --no-install-recommends --allow-unauthenticated  \
+    nvinfer-runtime-trt-repo-ubuntu1804-5.0.2-ga-cuda10.0 \
```
[the remark this answers is not in the document]
Comment (hsuw333) [reply tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu]: removed this

File: tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+ && apt-get install -y --no-install-recommends --allow-unauthenticated  \
+    libnvinfer7=${TF_TENSORRT_VERSION}-1+cuda11.1 \
```
[the remark this answers is not in the document]
Comment (hsuw333) [reply tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu:94]: yes, https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-722/install-guide/index.html#maclearn-net-repo-install
"cuda-x.x is CUDA version 10.2, 11.0, or 11.1 (for 11.2)"
will include this as comments in the file

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**